app/services/place_order.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+class PlaceOrder
+  def initialize(proxy_order, summarizer, logger, changes_loader)
+    @proxy_order = proxy_order
+    @subscription = proxy_order.subscription
+    @summarizer = summarizer
+    @logger = logger
+    @changes_loader = changes_loader
+  end
+
+  def call
+    return unless initialise_order
+
+    mark_as_processed
+
+    summarizer.record_order(order)
+    return summarizer.record_issue(:complete, order) if order.completed?
+
+    load_changes
+    return handle_empty_order if empty_order?
```

Comment:
TODO: move this empty case to the OrderWorkflow which will reduce coupling of this class and simplify it a lot more.